### PR TITLE
binutils: 2.35.1 -> 2.37

### PR DESCRIPTION
**Pull request by lovesegfault**
###### Motivation for this change
Hello binutils, my old friend
I've come to bump you again
Because a version softly creeping
Left its bugs while I was sleeping
And the patches that were planted in my brain
Still remains
Within the source of binutils

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/binutils/default.nix [hunk 1]
```diff
@@ -48,11 +49,6 @@ stdenv.mkDerivation {
     # Make binutils output deterministic by default.
     ./deterministic.patch
 
-    # Help bfd choose between elf32-littlearm, elf32-littlearm-symbian, and
-    # elf32-littlearm-vxworks in favor of the first.
-    # https://github.com/NixOS/nixpkgs/pull/30484#issuecomment-345472766
-    ./disambiguate-arm-targets.patch
-
```
Comment (lovesegfault) [inline pkgs/development/tools/misc/binutils/default.nix:55]: I'm not certain that this patch is no longer needed. I inspected the source and it _seems_ like it isn't, but a second opinion would be good.

**Comment by TredwellGit** [comment]
https://github.com/NixOS/nixpkgs/pull/118945

**Comment by lovesegfault** [comment]
God I hate GitHub so much

**Comment by TredwellGit** [comment]
This should target master while testing so it can be compared to https://hydra.nixos.org/jobset/nixpkgs/trunk.

**Comment by lovesegfault** [comment]
I'm just going to make a new PR, GitHub is just far too cursed

**Pull request closed without merging**